extension/src/cli/dvc/contract.ts

```diff
@@ -100,7 +100,7 @@ export type ExpWithError = {
 type ExpWithData = {
   rev: string
   name?: string
-  data: ExpData
+  data?: ExpData
```

Comment:
Our `DEFAULT_EXP_SHOW_OUTPUT` doesn't have `data` so data can possibly be undefined. This assumption about data being defined caused errors like this to be thrown:

```
rejected promise not handled within 1 second: TypeError: Cannot read properties of undefined (reading 'meta')
extensionHostProcess.js:105
stack trace: TypeError: Cannot read properties of undefined (reading 'meta')
	at hasCheckpoints (/Users/juliegalvan/Desktop/workspace/vscode-dvc/extension/dist/experiments/model/collect.js:191:29)
	at collectExperiments (/Users/juliegalvan/Desktop/workspace/vscode-dvc/extension/dist/experiments/model/collect.js:203:25)
	at ExperimentsModel.transformAndSet (/Users/juliegalvan/Desktop/workspace/vscode-dvc/extension/dist/experiments/model/index.js:55:140)
	at Experiments.<anonymous> (/Users/juliegalvan/Desktop/workspace/vscode-dvc/extension/dist/experiments/index.js:90:34)
	at Generator.next (<anonymous>)
	at fulfilled (/Users/juliegalvan/Desktop/workspace/vscode-dvc/extension/dist/experiments/index.js:5:58)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
```